application/workspace/v1_controller.py

```diff
@@ -72,7 +70,40 @@ def register_ws_v1(payload: WorkspaceRegisterRequest):
         LOG.error(f"Workspace registration failed. Error: {e}")
         return HttpError('Could not register new workspace'), 500
 
-    access_token = create_access_token(identity=payload.username)
-    refresh_token = create_refresh_token(identity=payload.username)
+    return WorkspaceCreateResponse(id=ws.id, name=ws.name, invitation_link_token=ws.invitation_link_token), 201
+
+
+@ws_v1.route("/<ws_id>/invitations", methods=["POST"])
+@jwt_required
+@json_convert(to=WorkspaceInvitationRequestItem)
+def invite_v1(payload: WorkspaceInvitationRequestItem, ws_id):
+
+    if WorkspaceUserModel.find(user_id=current_user.id, ws_id=ws_id) is None:
+        return HttpError('Could not find associated workspace'), 404
+
+    user = User.find(email=payload.email)
+    if user is None:
+        if WorkspaceInvitation.find(email=payload.email, ws_id=ws_id) is None:
+            WorkspaceInvitation(email=payload.email, ws_id=ws_id, start_date=datetime.datetime.now(), status=WorkspaceInvitationStatus.PENDING).save()
+    elif WorkspaceUserModel.find(user_id=user.id, ws_id=ws_id) is None:
+        # start_date = payload.start_date if payload.start_date else datetime.date.now()
+        start_date = datetime.datetime.now()
+        WorkspaceUserModel(user_id=user.id, ws_id=ws_id, start_date=start_date).save()
+
+    return "", 201
```

Comment:
I think I created a dataclass HttpMessage somewhere for such cases

Reply:
Not sure if we need to use HttpMessage with graphql

Reply:
This handler is not graphql, i think

Reply:
This handler is not graphql, i think

Reply:
It's not, but will remove once we have graphql mutation/resolver for that